backend/alert/tasks.py

```diff
@@ -25,20 +25,34 @@ def run_course_updates(semester=None):
 
 
 @shared_task(name="pca.tasks.send_alert")
-def send_alert(reg_id, sent_by=""):
-    result = Registration.objects.get(id=reg_id).alert(sent_by=sent_by)
+def send_alert(reg_id, sent_by="", close_notification=False):
+    result = Registration.objects.get(id=reg_id).alert(
+        sent_by=sent_by, close_notification=close_notification
+    )
     return {"result": result, "task": "pca.tasks.send_alert"}
 
 
 def get_active_registrations(course_code, semester):
     _, section = get_course_and_section(course_code, semester)
-    return list(section.registration_set.filter(notification_sent=False, deleted=False))
+    # Use the is_active filters statically defined in the Registration model
+    return list(section.registration_set.filter(**Registration.is_active_filter()))
+
+
+def get_registrations_for_alerts(course_code, semester, course_status="O"):
+    _, section = get_course_and_section(course_code, semester)
+    if course_status == "O":
+        return get_active_registrations(course_code, semester)
+    elif course_status == "C":
+        # Use the is_waiting_for_close_filter filters statically defined in the Registration model
+        return list(section.registration_set.filter(**Registration.is_waiting_for_close_filter()))
+    else:
+        return []
 
 
 @shared_task(name="pca.tasks.send_course_alerts")
-def send_course_alerts(course_code, semester=None, sent_by=""):
+def send_course_alerts(course_code, semester=None, sent_by="", course_status="O"):
```

Comment:
Putting a default status here makes me kind of nervous, since this is data we should be relying on the webhook for. It might be a bit more work to set a variable at every call site, but I think it would be worth it to make sure bugs don't pop up.

Reply:
Fixed